### ISSUE TITLE
Investigate making page load speed improvements on the server side

### ISSUE THREAD

**Issue by nickserv** [toroidal-code/codeflash#66]
I noticed that we are spending only 554ms loading resources for the home page (cache disabled on the client side), despite the entire pageload taking 1.42s. Rails is probably taking preparing its assets between HTTP requests.

We should probably either sit a cache in front of Rails or try to change how assets are set up on production. What do you guys think?

**Comment by toroidal-code** [comment]
Switch to Rubinius to increase concurrency.

**Comment by kristenmills** [comment]
No. Just no. Jruby is going to do more harm than good .

**Comment by nickserv** [comment]
Can we use JRuby to lengthen the command used to start the Puma server?

**Comment by kristenmills** [comment]
:+1:

**Comment by toroidal-code** [comment]
So I started using RBX instead of ruby, and it's a hell of a lot faster than YARV or  MRI, which have the 1 thread limit. Thoughts?

**Comment by nickserv** [comment]
But that doesn't work on Heroku right? :c

I'm totally up for RBX, but we'd need to be able to use it on production.

**Comment by toroidal-code** [comment]
Okay so Engine Yard uses RBX, my deploy server uses RBX, and this is shown to increase load speed. Now, we're also using NGINX to serve the static assets which further decreases load time. Anything else? Nick was talking about setting up a caching layer like Varnish.

**Comment by nickserv** [comment]
I think we should investigate how much performance gains we can get from caching at the NGINX level. If we don't like our results, we can think about sitting varnish in front of some of our servers.

**Comment by nickserv** [comment]
@toroidal-code Is this done?

**Comment by toroidal-code** [comment]
@thenickperson this is never done. Seriously though. We're doing pretty well right now with rbx, puma, nginx, and postgres, but we might want to start considering stuff like redis caching if we want to really fly. But for now we're doing well.

**Comment by nickserv** [comment]
Well, we already have good peformance improvements by using those technologies. I think we should just test to make sure assets aren't getting recompiled on the fly in production and then close this. We can continue making performance improvements for other things as we go along.

**Comment by toroidal-code** [comment]
look at the deploy system. that should tell you about the asset compilation. But I know it seems to be working right now

**Comment by nickserv** [comment]
Precompiling is working now, right? Should we close this?

**Comment by toroidal-code** [comment]
nah, i'm going to leave this open until we get some better server-side caching done, because something seems wonky with the rendering and serving on pages like /solutions.
